src/services/services.ts

```diff
@@ -2072,7 +2079,8 @@ namespace ts {
             compilationSettings: CompilerOptions,
             scriptSnapshot: IScriptSnapshot,
             version: string,
-            acquiring: boolean): SourceFile {
+            acquiring: boolean,
+            scriptKind?: ScriptKind): SourceFile {
 
             const bucket = getBucketForCompilationSettings(compilationSettings, /*createIfMissing*/ true);

```

Comment:
Discussed with Vlad - for the moment we won't support the above scenario. I'll add an assert with an appropriate message to make sure if this does occur it will be apparent what is going on.